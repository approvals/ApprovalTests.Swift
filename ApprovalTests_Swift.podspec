Pod::Spec.new do |s|
  s.name         = 'ApprovalTests_Swift'
  s.version      = '2.0.0'
  s.summary      = 'ApprovalTests for Swift, a powerful alternative to assertions'

  s.description = <<-DESC
                    Approval Tests are an alternative to assertions. Youâ€™ll find them useful for
                    testing objects with complex values (such as long strings), lots of properties,
                    or collections of objects.

                    ApprovalTests.Swift is compatible with the XCTest testing framework.
                  DESC

  s.homepage = 'https://github.com/approvals/ApprovalTests.Swift'
  s.license  = { :type => 'Apache License, Version 2.0', :file => 'LICENSE' }
  s.author   = { 'Jon Reid' => 'jon@qualitycoding.org' }
  s.social_media_url = 'https://twitter.com/qcoding'

  s.swift_version = '5.0'
  s.ios.deployment_target = '12.0'
  s.osx.deployment_target = '10.15'

  s.source = { :git => 'https://github.com/approvals/ApprovalTests.Swift.git', :tag => 'v2.0.0' }
  s.framework = 'XCTest'
  s.source_files      = "ApprovalTests.Swift/iOSApprovalsWatcher.py", "ApprovalTests.Swift/**/*.swift" 
  s.ios.exclude_files = "ApprovalTests.Swift/Reporters/macOS/*.swift"
  s.osx.exclude_files = "ApprovalTests.Swift/Reporters/iOS/*.swift"

end
