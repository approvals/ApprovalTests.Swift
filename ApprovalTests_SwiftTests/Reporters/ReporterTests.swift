#if os(iOS)
  import ApprovalTests_iOS
#else
  import ApprovalTests_Swift
#endif
import XCTest

final class ReporterTests: XCTestCase {
    func testFirstWorking() throws {
        let reporter1 = TestReporter(success: false)
        let reporter2 = TestReporter(success: true)
        let reporter3 = TestReporter(success: true)

        let success = FirstWorkingReporter(reporter1, reporter2, reporter3)
            .report(received: "r.text", approved: "a.text")

        XCTAssertTrue(success)
        XCTAssertEqual(reporter1.approved, "a.text")
        XCTAssertEqual(reporter2.approved, "a.text")
        XCTAssertEqual(reporter3.approved, "")
    }

    func testMultiReporter() throws {
        let reporter1 = TestReporter(success: true)
        let reporter2 = TestReporter(success: false)
        let reporter3 = TestReporter(success: true)

        let success = ReportWithEverything(reporter1, reporter2, reporter3)
            .report(received: "r.text", approved: "a.text")

        XCTAssertTrue(success)
        XCTAssertEqual(reporter1.approved, "a.text")
        XCTAssertEqual(reporter2.approved, "a.text")
        XCTAssertEqual(reporter3.approved, "a.text")
    }

    #if os(iOS)
        func test_nameWithSpaces() throws {
            let output = GenericDiffReporter.createCommandLine("/path with spaces/r.text", "/path with spaces/a.text", {
                received,
                approved in
                [received, approved]
            }, "diff")
            try Approvals.verify(output)
        }

        func test_nameWithoutSpaces() throws {
            let output = GenericDiffReporter.createCommandLine("/path_without_spaces/r.text", "/path_without_spaces/a.text", {
                received,
                approved in
                [received, approved]
            }, "diff")
            try Approvals.verify(output)
        }
    #endif

    #if os(OSX)
        func test_clipboardReporterOnMac() throws {
            let reporter = ReportMoveCommandToClipboard()

            let success = reporter.report(received: "/path with spaces/r.text", approved: "/path with spaces/a.text")

            XCTAssertTrue(success)
            let pasteboard = NSPasteboard.general
            let command = pasteboard.string(forType: .string)
            XCTAssertEqual(command, "mv \"/path with spaces/r.text\" \"/path with spaces/a.text\"")
        }
    #endif

    #if os(OSX)
        func test_fileLaunchingOnMac() throws {
            let reporter = ReportByOpeningReceivedFile()
            let process = reporter.makeProcess(received: "r.html", approved: "a.html")
            try Approvals.verify("\(process.executableURL!) \(process.arguments!)")
        }
    #endif
}
