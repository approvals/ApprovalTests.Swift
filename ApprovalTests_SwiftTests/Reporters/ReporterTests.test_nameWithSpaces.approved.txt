#!/bin/bash
diff "/path with spaces/r.text" "/path with spaces/a.text"
